scripts/maint/rename_c_identifier.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python3
+
+"""
+Helpful script to replace one or more C identifiers, and optionally
+generate a commit message explaining what happened.
+"""
+
+import argparse
+import fileinput
+import os
+import re
+import shlex
+import subprocess
+import sys
+import tempfile
+
+TOPDIR="src"
+
+def is_c_file(fn):
+    """
+       Return true iff fn is the name of a C file.
+
+       >>> is_c_file("a/b/module.c")
+       True
+       >>> is_c_file("a/b/module.h")
+       True
+       >>> is_c_file("a/b/module.c~")
+       False
+       >>> is_c_file("a/b/.module.c")
+       False
+       >>> is_c_file("a/b/module.cpp")
+       False
+    """
+    fn = os.path.split(fn)[1]
+    if fn.startswith("."):
+        return False
+    ext = os.path.splitext(fn)[1]
+    return ext in {".c",".h",".i","inc"}
+
+def list_c_files(topdir=TOPDIR):
+    """
+       Use git to list all the C files under version control.
+
+       >>> lst = list(list_c_files())
+       >>> "src/core/mainloop/mainloop.c" in lst
+       True
+       >>> "src/core/mainloop/twiddledeedoo.c" in lst
+       False
+    """
+    proc = subprocess.Popen(
+        ["git", "ls-tree", "--name-only", "-r", "HEAD", topdir],
+        stdout=subprocess.PIPE,
+        encoding="utf-8")
+    for line in proc.stdout.readlines():
+        line = line.strip()
+        if is_c_file(line):
+            yield line
+
+class Rewriter:
+    """
+       A rewriter applies a series of word-by-word replacements, in
+       sequence.  Replacements only happen at word boundaries.
+
+       >>> R = Rewriter([("magic", "secret"), ("words", "codes")])
+       >>> R.apply("The magic words are rambunctious bluejay")
+       'The secret codes are rambunctious bluejay'
+       >>> R.apply("The magical words are rambunctious bluejay")
+       'The magical codes are rambunctious bluejay'
+       >>> R.get_count()
+       3
+    """
+    def __init__(self, replacements):
+        """Make a new Rewriter. Takes a sequence of pairs of
+           (from_id, to_id), where from_id is an identifier to replace,
+           and to_id is its replacement.
+        """
+        self._patterns = [ ]
+        for id1,id2 in replacements:
```

Comment:
nit: `id1, id2` unless you are really okay with missing a space

Reply:
2be5f50bf9c8f1c7e8dfeeac1f5ea605e7fed3e8